### PR TITLE
MODLD-367: Remove DataFields that do not have subfields from the MARC record

### PR DESCRIPTION
**Pull request by pkjacob**

### CODE REVIEW

File: src/main/java/org/folio/marc4ld/service/ld2marc/resource/Resource2MarcRecordMapperImpl.java [hunk 1]
```diff
@@ -51,12 +51,17 @@ public Record toMarcRecord(Resource resource) {
...
       .filter(mapper -> mapper.canMap(edge.getPredicate(), edge.getTarget()))
       .findFirst()
       .map(mapper -> mapper.map(edge.getTarget()))
       .map(List::of)
       .orElseGet(() -> getFields(cfb, edge));
+
+    return dataFields
+      .stream()
+      .filter(df -> isNotEmpty(df.getSubfields()))
+      .toList();
```
Comment (aleksei-pronichev) [inline src/main/java/org/folio/marc4ld/service/ld2marc/resource/Resource2MarcRecordMapperImpl.java:64]: just simplify
```java
return ld2MarcMappers.stream()
      .filter(mapper -> mapper.canMap(edge.getPredicate(), edge.getTarget()))
      .findFirst()
      .map(mapper -> mapper.map(edge.getTarget()))
      .map(List::of)
      .orElseGet(() -> getFields(cfb, edge))
      .stream()
      .filter(df -> isNotEmpty(df.getSubfields()))
      .toList();
```

File: src/test/java/org/folio/marc4ld/mapper/Bibframe2Marc010IT.java [hunk 2]
```diff
@@ -69,6 +69,22 @@ void should_map_all_subfields_z() {
...
+      lccn("11111111", "invalid_status")
+    );
+
+    // when
+    var result = bibframe2MarcMapper.toMarcJson(resource);
+
+    // then
+    assertThat(result)
+      .isEqualTo(expectedMarc);
+  }
```
Comment (aleksei-pronichev) [inline src/test/java/org/folio/marc4ld/mapper/Bibframe2Marc010IT.java:86]: Empty Marc is too simple an object to keep separately.
suggest to replace to a simple assertion, like this
```
    assertThat(result.getfields)
      .isEmpty
```
It is easier to support

**Review by aleksei-pronichev — APPROVED** [review]

**Review by PBobylev — APPROVED** [review]

File: src/test/java/org/folio/marc4ld/mapper/Bibframe2Marc010IT.java (hunk 2 above)
Comment (pkjacob) [reply src/test/java/org/folio/marc4ld/mapper/Bibframe2Marc010IT.java:92]: Done

File: src/main/java/org/folio/marc4ld/service/ld2marc/resource/Resource2MarcRecordMapperImpl.java (hunk 1 above)
Comment (pkjacob) [reply src/main/java/org/folio/marc4ld/service/ld2marc/resource/Resource2MarcRecordMapperImpl.java:62]: Done

**Pull request merged**